### ISSUE TITLE
if using MSI, failover via HA doesn't work if existing VNET is in a different RG

### ISSUE THREAD

**Issue by mikeoleary** [F5Networks/f5-azure-arm-templates#177]
When deploying to existing stack where VNET is in a different ResourceGroup than the BIG-IP, HA failover calls fail with errors indicating the permissions required on the existing VNET are not present.

Error from /var/log/restnoded/restnode.log:

Fri, 17 Apr 2020 15:32:41 GMT - severe: [f5-cloud-failover] failover.execute() error: The client '...' with object id '...' has permission to perform action 'Microsoft.Network/networkInterfaces/write' on scope '/subscriptions/.../resourceGroups/molearyfri-compute/providers/Microsoft.Network/networkInterfaces/molearyfrivm-ext0'; however, it does not have permission to perform action 'Microsoft.Network/virtualNetworks/subnets/join/action' on the linked scope(s) '/subscriptions/.../resourceGroups/molearyfri-networkrg/providers/Microsoft.Network/virtualNetworks/molearyfri-vnet/subnets/external' or the linked scope(s) are invalid.

## Workaround
The roleAssignment created in template indicates the scope is only the ResourceGroup to which we are deploying. I was able to get around the below error only by manually creating a roleAssignment of Contributor for the 2x VM instances in the RG that contained the existing VNET. After this, failover of floating IP address was successful.

## Template
This may apply to all HA failover templates using CFE and MSI's, but this was the template that I used for testing.
https://github.com/F5Networks/f5-azure-arm-templates/tree/master/supported/failover/same-net/via-api/n-nic/existing-stack/payg

## Severity Level
4

**Comment by shyawnkarim** [comment]
This is a known issue and is currently in our backlog. Internal ID AUTOSDK-302.

**Comment by JeffGiroux** [comment]
One possible workaround edit would be in the Microsoft.Authorization/roleAssignments. This would link the role to scope level of subscription instead of RG.

old
```
"scope": "[resourceGroup().id]"
```

new
```
"scope": "[subscription().id]"
```

Reference: https://github.com/Azure/azure-quickstart-templates/tree/master/subscription-level-deployments/subscription-role-assigment

**Comment by mikeoleary** [comment]
Thanks @JeffGiroux - I found just editing the scope of the roleAssignment did not work. You would then need to make your deployment at the subscription level, and not at the RG level. Otherwise I am getting this error if I edit the scope only:

"InvalidCreateRoleAssignmentRequest" with message "The request to create role assignment '[guid]' is not valid. Role assignment scope '/subscriptions/[guid]' must match the scope specified on the URI '/subscriptions/[guid]/resourcegroups/[rgname]'"

So this workaround may work but it would require the deployment be done at subscription, not ResourceGroup, level.

I think you could also add a 2nd roleAssignment and set the scope to be the RG in which the existing VNET is deployed. However, to deploy a resource to a different RG than your current deployment, you'd need a nested template and then to just deploy the additional roleAssignments to the 2nd RG in that nested template. I think that would work, however I'm personally going to just use the manual workaround outlined above and wait for this backlog item to be addressed.

**Comment by shyawnkarim** [comment]
Closing. This issue was resolved with [Cloud Failover Extension Release 1.3.0](https://github.com/F5Networks/f5-cloud-failover-extension/releases/tag/v1.3.0).
